xtask/src/build_docker_image.rs

```diff
@@ -248,10 +256,14 @@ pub fn build_docker_image(
         if no_cache {
             docker_build.arg("--no-cache");
         } else if engine.kind.supports_cache_from_type() {
-            docker_build.args([
-                "--cache-from",
-                &format!("type=registry,ref={base_name}:main"),
-            ]);
+            if let Some(ref cache_from) = cache_from {
+                docker_build.args(["--cache-from", cache_from]);
+            } else {
+                docker_build.args([
+                    "--cache-from",
+                    &format!("type=registry,ref={base_name}:main"),
```

Comment:
would be nice if we could have some string replacements here so that it's possible to express the same thing the `format!` does here

that way we could also make the arguments have defaults, so something like

```rs
    /// Cache from option for docker, would only work if engine supports cache from type and no_cache is not set to true
    #[clap(long, default_value = "type=registry,ref={base_name}:main")]
    pub cache_from: String,
```

Reply:
mhmm, problem is, format! won't accept anything that is not string literal - it is a macros and it wants to see the actual string in compile time. Assuming I understood correctly what you are asking, I have a feeling that `format!` will not allow it

Reply:
I guess I can do something like `&cache_from.replace("{base_name}", &base_name)`

Reply:
Though, I am not sure if in this case cache-to should also support such "templated" input or not

Reply:
Ok, I have unified how --cache-from is passed - now instead of format!, which would not allow dynamic template, I am simply using `replace` and cache-from comes with default that maches usual behavior. From another side, cache-to remained optional, however when specified, same replacement would take place (not sure if necessary or not TBH, but just to make behavior consistent for two options). 

This is commit: 
https://github.com/cross-rs/cross/pull/1248/commits/0a157e58188ceec08d4f4f6c5ff3627f3fb1e292